pkg/knowledge_base2/template_kb_mock_test.go

```diff
@@ -0,0 +1,257 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: ./kb.go
+//
+// Generated by this command:
+//
+//	mockgen --source=./kb.go -destination=./template_kb_mock_test.go -package=knowledgebase2
+//
+
+// Package knowledgebase2 is a generated GoMock package.
+package knowledgebase2
+
+import (
+	reflect "reflect"
+
+	graph "github.com/dominikbraun/graph"
+	construct2 "github.com/klothoplatform/klotho/pkg/construct2"
+	gomock "go.uber.org/mock/gomock"
+)
+
+// MockTemplateKB is a mock of TemplateKB interface.
+type MockTemplateKB struct {
```

Comment:
Does this replace the `MockKB` in `kbtesting`?

Reply:
The reason the mock was put in that package (and without the `_test.go` suffix) was so that it could be used by tests outside of the package.